SmartDeviceLink/private/SDLChoiceSetManager.m

```diff
@@ -445,7 +447,27 @@ - (void)dismissKeyboardWithCancelID:(NSNumber<SDLInt> *)cancelID {
 /// @param choices The choices to be uploaded
 /// @return The choices that have not yet been uploaded to the module
 - (NSSet<SDLChoiceCell *> *)sdl_choicesToBeUploadedWithArray:(NSArray<SDLChoiceCell *> *)choices {
-    NSMutableSet<SDLChoiceCell *> *choicesSet = [NSMutableSet setWithArray:choices];
```

Comment:
It's important to keep this as a mutable set to weed out duplicates first. I think you can use fast enumeration to loop a set and I don't think the index is important here:

```objc
for (SDLChoiceCell *cell in choicesSet) {

}
```